lib/client-assets.php

```diff
@@ -662,3 +662,40 @@ function gutenberg_extend_block_editor_settings_with_fse_theme_flag( $settings )
 	return $settings;
 }
 add_filter( 'block_editor_settings', 'gutenberg_extend_block_editor_settings_with_fse_theme_flag' );
+
+/**
+ * Sets the editor styles to be consumed by JS.
+ */
+function gutenberg_extend_block_editor_styles_html() {
+	$handles = array(
+		'wp-block-editor',
+		'wp-block-library',
+		'wp-edit-blocks',
+	);
+
+	$block_registry = WP_Block_Type_Registry::get_instance();
+
+	foreach ( $block_registry->get_all_registered() as $block_name => $block_type ) {
+		if ( ! empty( $block_type->style ) ) {
+			$handles[] = $block_type->style;
+		}
+
+		if ( ! empty( $block_type->editor_style ) ) {
+			$handles[] = $block_type->editor_style;
+		}
+	}
+
+	$handles = array_unique( $handles );
+	$done    = wp_styles()->done;
+
+	ob_start();
+
+	wp_styles()->done = array();
+	wp_styles()->do_items( $handles );
+	wp_styles()->done = $done;
+
+	$editor_styles = wp_json_encode( array( 'html' => ob_get_clean() ) );
+
+	echo "<script>window.__editorStyles = $editor_styles</script>";
```

Comment:
Is there a better to do that without the new variable. Maybe by just reusing the "styles" property in the editor settings?

Reply:
Yeah, I can look into that

Reply:
I tried to do this, but the styles are not available yet at the time when the editor scripts are registered.

Reply:
I  was not suggesting at the script registration, I was more suggesting `block_editor_settings` filter for post editor and `gutenberg_edit_site_init` for the site editor.

Reply:
That's what I mean. The `block_editor_settings` filter is run at script registration, which is too early. The settings are passed as an inline script initialising the editor.